libglusterfs/src/common-utils.c

```diff
@@ -3038,8 +3038,19 @@ gf_set_volfile_server_common(cmd_args_t *cmd_args, const char *host,
     }
 
     INIT_LIST_HEAD(&server->list);
+    server->port = port;
 
-    server->volfile_server = gf_strdup(host);
+    char *duphost = gf_strdup(host);
+    char *lastptr = rindex(duphost, ':');
+    if (lastptr) {
+        *lastptr = '\0';
+        long port_argument = strtol(lastptr + 1, NULL, 0);
```

Comment:
This can detect last part of IPv6 element as Port? For web URLs, IPv6 is used as `http://[<ipv6-address>]:port`.  (Python urllib parse example: https://github.com/python/cpython/blob/main/Lib/urllib/parse.py#L202-L203)

Reply:
Yes, there are some challenges here. Hence I want to add test cases, without which would be hard for me to test myself. Expect some delay in this.